src/common/merge.ts

```diff
@@ -67,7 +67,11 @@ export const mergeWithRetry = async (
     /* eslint-disable-next-line @typescript-eslint/no-base-to-string */
     logDebug(`Original error: ${(error as Error).toString()}.`);
 
-    if (shouldRetry(error) && retryCount <= maximumRetries) {
+    if (shouldRetry(error) === false) {
+      return;
```

Comment:
Hm, should we put the `logInfo` and `logDebug` above inside this branch? It doesn't seem to make much sense to log it all the time?

Reply:
This information is still useful if it fails the last time. I changed the logic to this: a59bda6153358fc33125e1f701e2a5ff21b2806d WDYT?